docs/literate/make.jl

```diff
@@ -0,0 +1,122 @@
+using Literate
+using Test
+using HTTP
+import Pkg
+
+# Function to create markdown and notebook files for specific file
+function create_files(title, file; folder="")
+    if folder == ""
+        notebook_filename = "$(splitext(file)[1]).ipynb"
+    else
+        notebook_filename = "$folder/$(splitext(file)[1]).ipynb"
+    end
+
+    binder_badge   = "# [![]($binder_logo)]($binder_url/$notebook_filename)"
+    nbviewer_badge = "# [![]($nbviewer_logo)]($nbviewer_url/$notebook_filename)"
+    download_badge = "# [![]($download_logo)]($download_url/$notebook_filename)"
+    
+    # Generate notebooks
+    function preprocess_notebook(content)
+        return string("# # $title\n\n", preprocess_links(content))
+    end
+    Literate.notebook(joinpath(repo_src, folder, file), joinpath(notebooks_dir, folder); execute=false, preprocess=preprocess_notebook, credit=false)
+
+    # Generate markdowns
+    function preprocess_docs(content)
+        return string("# # [$title](@id $(splitext(file)[1]))\n $binder_badge\n $nbviewer_badge\n $download_badge\n\n", preprocess_links(content))
+    end
+    Literate.markdown(joinpath(repo_src, folder, file), joinpath(pages_dir, folder); preprocess=preprocess_docs,)
+
+    @testset "TrixiTutorials $title" begin include(joinpath(repo_src, folder, file)) end
```

Comment:
Here, the tutorial jl source file are called to check for errors. Could be moved to the test folder as well.

Reply:
@ranocha What do you think about calling the tests outside `create_files()` with something like that
```
@testset "TrixiTutorials" begin
        for (i, (title, filename)) in enumerate(files)
            if filename isa Vector # Several files of one topic
                for j in eachindex(filename)
                    @testset "$(filename[j][2][2])" begin include(joinpath(repo_src, filename[j][2][1], filename[j][2][2])) end
                end
            else # Single files
                @testset "$title" begin include(joinpath(repo_src, filename)) end
            end
        end
    end
```
It's a bit more code, but all the test output is together and not scattered.

Reply:
Sounds good to me